.github/workflows/testTKG.yml

```diff
@@ -4,10 +4,11 @@ on:
     types: [created]
 jobs:
   testTKG:
+    permissions: write-all
```

Comment:
Let's not turn on write permissions for everything, as we only need it for a few things from the list https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#defining-access-for-the-github_token-scopes 

Also, I do not think you can test a Github actions change by using the workflow itself, we will likely just have to make a precise change, merge it, then check if it is behaving as we want (otherwise test it in a separate personal repo).

Reply:
Tested within the personal forked repo Actions and its working fine. So likely a permission issue when PR from forked repository is tested from main repo.